src/index.tsx

```diff
@@ -367,24 +255,59 @@ class App extends React.Component<{}, AppState> {
 
   public state: AppState = {
     draggingElement: null,
+    elements: [],
     elementType: "selection",
     exportBackground: false,
     exportVisibleOnly: true,
     exportPadding: 10
   };
 
+  clearSelection = () => {
+    const newElements = [...this.state.elements];
+    newElements.map(element => {
+      element.isSelected = false;
```

Comment:
This is still mutating. You need to write

```js
return {...element, isSelected: false}
```

Reply:
Yup, there are probably more cases like this one. I tried to find them all, but the index file is big :p

Reply:
You can search for `isSelected =`